### PR TITLE
BAU: Optimize and cleanup some tests

### PR DESCRIPTION
**Pull request by jakubmiarka**
With an introduction of parallel testing we can optimize the runtime even more by
optimizing some of the tests and spreading the tests across features.
Instead of having a couple chunk feature test, it's better to break them
down to multiple and smaller bits.
I also removed some of the redundant tests which are being tested somewhere else.

### CODE REVIEW

File: features/account_creation.feature [hunk 1]
```diff
@@ -104,59 +104,3 @@ Feature: User account creation
       | surname        | Doe            |
       | dateofbirth    | 1987-03-03     |
       | currentaddress | 123 Test Drive |
-
-  Scenario: Sign in without cycle 3 and unsigned by hub
```
Comment (jakubmiarka) [inline features/account_creation.feature]: moved to `simple_sign_in.feature`

File: features/journey_hint.feature [hunk 2]
```diff
@@ -21,33 +21,3 @@ Scenario: Journey hint Registration
     And they select journey hint "Sign-in with eIDAS"
     And they start a journey
     Then they should arrive at the country picker page
-
-  Scenario: Journey hint Sign-in with eIDAS when Eidas is Disabled
```
Comment (jakubmiarka) [inline features/journey_hint.feature]: all of this are unit tests pretty much covered in https://github.com/alphagov/verify-frontend/blob/master/spec/features/redirects_with_journey_hint_authn_request_spec.rb

File: features/pause_and_resume.feature [hunk 3]
```diff
@@ -54,55 +54,6 @@ Feature: Pause and Resume User Journey
     When they give their consent
     Then they should be successfully verified
 
-  Scenario: User pauses and resumes from an IDP e-mail link, on same device, and resumes registration
```
Comment (chrisclayson) [inline features/pause_and_resume.feature]: Did you mean to move these to another feature or are you just removing them?

Comment (jakubmiarka) [inline features/pause_and_resume.feature]: covered in here: https://github.com/alphagov/verify-frontend/blob/master/spec/features/user_visits_paused_page_spec.rb

Comment (jakubmiarka) [reply features/pause_and_resume.feature]: added a comment just now ;)

Comment (chrisclayson) [reply features/pause_and_resume.feature]: No, thats OK ... I see your point!

**Pull request merged**